### PR TITLE
🌱 add broken link checker action

### PR DESCRIPTION
**Pull request by tuminoid**
Take broken link checker action from CAPM3 and add it here as well.

### CODE REVIEW

**Review by kashifest — APPROVED** [review]

File: .markdownlinkcheck.json [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+        "headers": {
+            "Accept-Encoding": "zstd, br, gzip, deflate"
+        }
+    }],
+    "timeout": "10s",
+    "retryOn429": true,
+    "retryCount": 5,
+    "fallbackRetryDelay": "30s",
+    "aliveStatusCodes": [200, 206]
+}
```
Comment (zhouhao3) [inline .markdownlinkcheck.json:17]: Seems good to me, but there is a problem needs to be solved.

Comment (tuminoid) [reply .markdownlinkcheck.json:17]: Sorry, which problem are you referring to?

Comment (zhouhao3) [reply .markdownlinkcheck.json:17]: I saw a `No newline at end of file` prompt on the page, it seems that the last line lacks a newline character.

**Review by smoshiur1237 — APPROVED** [review]
LGTM

**Review by kashifest — APPROVED** [review]

**Pull request merged**